### ISSUE TITLE
With DataFrame colums tab does not work

### ISSUE THREAD

**Issue by hein123** [jupyter/notebook#2883]
Don't know if I updated something wrong, however when creating pandas and trying see list of functions with a dataframe, it creates the following:

df['col2']. --> press tab --> df['col2'],ipynb_checkpoints/

should I reinstall?

**Comment by takluyver** [comment]
No, i think that's a known limitation of our tab completion system, so reinstalling won't help.

If the column names are valid Python identifiers, you can access them like `df.col2`, which should allow tab completion to work.

**Comment by pavelpe** [comment]
> No, i think that's a known limitation of our tab completion system, so reinstalling won't help.
> 
> If the column names are valid Python identifiers, you can access them like `df.col2`, which should allow tab completion to work.

But according to this blog [https://medium.com/@willgdjones/installing-jupyterlab-1017fa6c1615](url)
it is not just pandas issue. It supposed to work (and it works) even on cases like:
`(3+1).`

According to him, the issue was with the Jupyter Notebook version and upgrading it solved the issue.
But I have latest Notebook version and this problem still happens.
![image](https://user-images.githubusercontent.com/33594067/48857488-ac4d0580-edc1-11e8-9b06-63e28f4ec33d.png)

While doing:
`x = (3 + 1)`
`x.`
gives the expected auto-completion result (**w/o even running this cell** ):
![image](https://user-images.githubusercontent.com/33594067/48857691-2b423e00-edc2-11e8-8543-d946df88d199.png)

I really hope this Tab Auto Completion issue will be solved in near future.
